src/services/completions.ts

```diff
@@ -2180,6 +2185,23 @@ export function getCompletionEntriesFromSymbols(
     }
 }
 
+/**
+ * When filling completions for value-time locations in JS files, we'll want
+ * to only consider symbols that seem to have a value declaration. If a
+ * symbol no known declarations we cautiously include them just to be safe.
+ */
+function symbolMayHaveValueDeclaration(symbol: Symbol): boolean {
+    return !symbol?.declarations?.length || symbol.declarations.some(declaration => {
+        switch (declaration.kind) {
+            case SyntaxKind.InterfaceDeclaration:

```

Comment:
What about namespaces? i also avoid them in my own code so I don't know the exact rules - but it feels like they can be type-only, right? 🤔

Reply:
It really feels like this should be using the mentioned `SymbolFlags.Value` if that's already precalculated for us. Do you know why it doesn't currently work as expected?

Reply:
Namespaces can be type-only or not. The non-type-only namespace symbols are marked with SymbolFlags.ValueModule if I recall correctly.

Reply:
FWIW I do intend on investigating this more deeply - just haven't had the consecutive time to do so yet. In theory, +1 on going by `SymbolFlags`!